module/Decision/src/Decision/Controller/FileBrowser/LocalFileReader.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+
+namespace Decision\Controller\FileBrowser;
+
+/**
+ * Description of LocalFileReader
+ *
+ * @author s134399
+ */
+class LocalFileReader implements FileReader
+{
+
+    /**
+     * The location in the local filesystem that is considered the 'root' for this browser
+     * @var string
+     */
+    static private $root;
+
+    public function __construct($root)
+    {
+        $this->root = $root;
+    }
+
+    public function downloadFile($path)
+    {
+        $fullPath = $this->root . $path;
+        if (!is_file($fullPath)) {
+            return null;
+        }
+        $response = new \Zend\Http\Response\Stream();
+        $response->setStream(fopen('file://' . $fullPath, 'r'));
+        $response->setStatusCode(200);
+        $headers = new \Zend\Http\Headers();
+        $headers->addHeaderLine('Content-Type', 'application/octet-stream')
+                ->addHeaderLine('Content-Disposition', 'attachment; filename="' . end(explode('/', $fullPath)) . '"')
+                ->addHeaderLine('Content-Length', filesize($fullPath));
+        $response->setHeaders($headers);
+        return $response;
+    }
+
+    public function listDir($path)
+    {
+        //remove the trailing slash from the dir
+        $fullPath = $this->root . $path;
+        if (!is_dir($fullPath)) {
+            return null;
+        }
+        //We can insert an additional /, except when when $path is the root
+        $delimiter = $path !== '' ? '/': '';
```

Comment:
When you fix the stuff mentioned by @jszanto, lease also add a space before the `:` ;)

Reply:
@jszanto Why do we do the codeclimate thing again? That ghastly checker nags on and on about so-called 'cognitive complexity' (the mere assumption something with that name could be detected automatically by an entity that has no cognition is a logic fallacy). But it ignores simply stylistic stuff like here!